packages/daemon/src/daemon.js

```diff
@@ -404,6 +411,82 @@ const makeEndoBootstrap = async (
           }))(),
         internal: undefined,
       };
+    } else if (formula.type === 'handle') {
+      context.thisDiesIfThatDies(formula.target);
+      return {
+        external: {},
+        internal: {
+          targetFormulaIdentifier: formula.target,
+        },
+      };
+    } else if (formula.type === 'endo') {
+      /** @type {import('./types.js').FarEndoBootstrap} */
+      const endoBootstrap = Far('Endo private facet', {
+        // TODO for user named
+        ping: async () => 'pong',
+        terminate: async () => {
+          cancel(new Error('Termination requested'));
+        },
+        host: () => {
+          // Behold, recursion:
+          return /** @type {Promise<import('./types.js').EndoHost>} */ (
+            // eslint-disable-next-line no-use-before-define
+            provideValueForFormulaIdentifier(formula.host)
+          );
+        },
+        leastAuthority: () => {
+          // Behold, recursion:
+          return /** @type {Promise<import('./types.js').EndoGuest>} */ (
+            // eslint-disable-next-line no-use-before-define
+            provideValueForFormulaIdentifier(formula.leastAuthority)
+          );
+        },
+        webPageJs: () => {
+          if (formula.webPageJs === undefined) {
+            throw new Error('No web-page-js formula provided.');
+          }
+          // Behold, recursion:
+          // eslint-disable-next-line no-use-before-define
+          return provideValueForFormulaIdentifier(formula.webPageJs);
+        },
+        importAndEndowInWebPage: async (webPageP, webPageNumber) => {
+          const { bundle: bundleBlob, powers: endowedPowers } =
+            /** @type {import('./types.js').EndoWebBundle} */ (
+              // Behold, recursion:
+              // eslint-disable-next-line no-use-before-define
+              await provideValueForFormulaIdentifier(
+                `web-bundle:${webPageNumber}`,
+              ).catch(() => {
+                throw new Error('Not found');
+              })
+            );
+          const bundle = await E(bundleBlob).json();
+          await E(webPageP).makeBundle(bundle, endowedPowers);
+        },
+      });
+      return {
+        external: endoBootstrap,
+        internal: undefined,
+      };
+    } else if (formula.type === 'least-authority') {
+      /** @type {import('./types.js').EndoGuest} */
+      const leastAuthority = Far('EndoGuest', {
+        async request() {
+          throw new Error('declined');
+        },
+      });
+      return { external: leastAuthority, internal: undefined };
+    } else if (formula.type === 'pet-store') {
+      const external = petStorePowers.makeIdentifiedPetStore(
+        formulaNumber,
+        assertPetName,
+      );
+      return { external, internal: undefined };
+    } else if (formula.type === 'pet-inspector') {
```

Comment:
We could probably turn this would-be-switch block into a JavaScript map from type to behavior now.